peer.go

```diff
@@ -1521,9 +1525,23 @@ out:
 				continue
 			}
 
+			// We'll query the localChanCfg of the new channel to
```

Comment:
Why not just fetch the _actual_ policy here, rather than mostly using the default policy? As is, it's still incorrect as we'll add a link with the wrong policy compared to what _may_ have been advertised to the network.

Reply:
When creating the initial channel update, the default fees are used + the `min_htlc` from the `locaChanCfg`. Rationale for not using what has been advertised to the network (by querying the graph database) is that there's a potential race here: 

We don't send out our `ChannelUpdate` before we have sent the `fundingLocked` message, while we add the link to the switch after receiving the remote's `fundingLocked`. So at this point there's no guarantees it is in the graph.

I would be fine with querying the graph, and if the policy is not found, falling back to what we already have here. My main concern about this is that I think this fallback would be executed quite often, so we should instead make sure it will reflect the actual values that are (or might not yet) advertised.

Reply:
> So at this point there's no guarantees it is in the graph.

If the update goes out _before_ this in any case, then we have the same situation, no? All info can only be _eventually_ consistent so seems like in either case, we end up with the same set of ambiguity.

Reply:
> You mean the newChanReq can carry this information? That sounds like a good idea.
Yep the gossiper should still have access to his info, as is _just_ created the chan update to actually send out.

Reply:
> Yep the gossiper should still have access to his info, as is just created the chan update to actually send out.

The gossiper might not have this information yet, as the `ChannelUpdate` might not have been created, as mentioned above.

I started implementing the carrying the `baseFee`, `feeRate`, `timelock` and `minHtlc` as part of the `newChanReq`, but realized this will be a bigger change. Since we never store the fees anywhere during the funding process, we don't have direct access to them when receiving the `fundingLocked` message (at this point we add the channel to the switch). 

To reliably persist such custom fees during restarts we would need to add them to the database (`OpenChannel` mist likely), which seems a bit overkill as we are only using default fees at this point. The `min_htlc` value is already stored in `OpenChannel` as part of the `localChanCfg` and will therefore be persisted.

Reply:
Gotcha make sense, only min_htlc is exposed atm, so a user can't create a channel with any other field deviating _but_ this one.